service/surf/apps/core/middleware.py

```diff
@@ -0,0 +1,44 @@
+import ipaddress
+
+from django.conf import settings
+from django.urls import resolve
+from django.core.exceptions import PermissionDenied
+
+
+def get_request_ips(request):
```

Comment:
This is the description of the attack/problem: https://www.djangoproject.com/weblog/2009/jul/28/security/#secondary-issue
Django basically is not supporting what we're trying to do, because they think it is not possible to provide a generic secure solution. We are behind a trusted load balancer, so I think we can still pull it off.

Reply:
@fako and @Ruben-Hartog another option might be to shift the blocking of IP ranges to the AWS ALB. Then we are always sure we are dealing with the correct source IP address. I've taken a quick look at the documentation and I think it's possible to create a rule to only allow request to /admin and /admin/* if the source IP is from a specific range.

Reply:
@fako adding https://github.com/un33k/django-ipware adds some complexity I'm not sure we need. If I understand correctly it adds a list of preferred headers to look for the "real source ip" (eg. X-Forwarded-For). What happens if this header is already set by the requester? Does AWS overwrite the header?

Reply:
It does more than one thing. I would go for setting "trusted_proxy_ips" to the LB IP. Then it checks if the IP comes from a trusted source. Not sure if AWS is actually specifying itself as being a proxy though. That doesn't show from the documentation :(

Reply:
Happy with putting it on a load balancer. Means there are no tests, but if we all agree we can live with that I'm happy.